### PR TITLE
Refactoring training and inference script to match source code changes

### PR DESCRIPTION
**Pull request by vratins**
Changes

scripts/train.py 
  - Replace --use_slae flag with --encoder_type choices: gvp, slae, esm
  - Add dataset quality check args: --max_com_dist, --max_clash_fraction, --clash_dist, --interface_dist_threshold, --min_water_residue_ratio
  - Add per-water filtering args: --max_protein_dist, --min_edia, --max_bfactor_zscore with toggle flags
  - Add scheduler args: --scheduler, --warmup_steps, --eta_min_factor, --step_size, --step_gamma
  - Add model args: --slae_dim, --esm_dim, --grad_accum_steps, --drop_rate, --n_message_gvps, --n_update_gvps
  - New helpers: _extract_quality_config(), _build_dataset_config(), _required_embedding_field(), _resolve_embedding_dim(), resolve_encoder_config(), build_scheduler()
  - Training loop now supports gradient accumulation and dual schedulers (warmup + main)
  - Save resolved_encoder_config in config.json for inference compatibility

scripts/inference.py (+185 lines)
  - Import NUM_RBF from src.constants
  - Add --geometry_cache CLI arg to override model's geometry cache
  - Update build_model_from_config() to use resolved_encoder_config when available
  - Pass encoder_type, geometry_cache_name, include_mates to dataset

Note: A lot of the changes here are conditioned on changes in the source code that are part of other PRs

### CODE REVIEW

File: scripts/inference.py [hunk 1]
```diff
@@ -253,13 +326,15 @@ def run_inference_batch(
...
-            })
+            results.append(
+                {
+                    "protein_pos": graph["protein"].pos.numpy(),
+                    "water_true": graph["water"].pos.numpy(),
+                    "water_pred": water_pred,
+                    "trajectory": None,
+                    "pdb_id": getattr(graph, "pdb_id", None),
+                }
+            )
```
[the remark this answers is not in the document]
Comment (vratins) [reply scripts/inference.py]: Yep, fixed this

File: scripts/train.py [hunk 2]
```diff
@@ -26,52 +46,256 @@
...
+        action="store_true",
+        default=True,
+        help="Keep workers alive between epochs",
+    )
+    p.add_argument(
+        "--no_persistent_workers",
+        dest="persistent_workers",
+        action="store_false",
+        help="Disable persistent_workers",
+    )
```
[the remark this answers is not in the document]
Comment (vratins) [reply scripts/train.py]: Yep, simplifying this
